### PR TITLE
chore: notification telemetry 

### LINKED ISSUE #26785: Capture notification telemetry
We want to collect the following information about notifications:

- During the "check for updates" call, we want to record the current device settings under "Desktop Notifications" by sending an extra header with this information
- When the debug page is viewed, we should send a `source` attribute noting whether or not the view was from the user clicking a notification

Follow the requirements in the [Telemetry section of the ClickUp](https://app.clickup.com/18033298/docs/h6amj-43107/h6amj-41927?block=block-0a4667f5-7723-42a1-ae5f-eb40855e503b)

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes #26785

### Additional details

Capture some non-intrusive and non identifying telemetry for App Notifications.

### Steps to test

There are two ways a user might navigate to the Debug page:

- via side navigation
- via a notification

We want to capture how the user is navigating to the Debug page. That's what you are testing here.

The easiest way to do this is to turn on debugging for `cypress:data-context:actions:EventCollectorActions` and look for a message that says `Recorded machine-linked` with the expected event info.

Then you can trigger the event like this:

1. `source: sidebar` - just click the Debug icon in from the App. It will launch a mutation that eventually calls the `recordEvent` action.
2. `source: notification`. This is a bit more tricky. You'll need to trigger a notification. You can see how to do that by looking at [this PRs test plan](https://github.com/cypress-io/cypress/pull/26974). Once you click a notification, it should also record the event with the correct `medium`.

Example:

image

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/app/cypress/e2e/debug.cy.ts [hunk 1]
```diff
@@ -58,9 +58,24 @@ describe('App - Debug Page', () => {
 
     cy.visitApp()
 
+    cy.intercept('mutation-SideBarNavigation_RecordEvent').as('recordEvent')
     cy.findByTestId('sidebar-link-debug-page').click()
     cy.findByTestId('debug-container').should('be.visible')
 
+    // TODO: comment back in once https://github.com/cypress-io/cypress/issues/26693 merges
```
Comment (lmiller1990) [inline packages/app/cypress/e2e/debug.cy.ts]: This entire test is really flaky locally, even before this addition. I believe this is improved significantly in #26693.

**Comment by warrensplayer** [comment]
This looks like it will take some sort of Cloud work to implement.  Please see internal Slack initiative channel for more details. For now, moving this back to 'draft' state.

File: packages/data-context/src/sources/VersionsDataSource.ts [hunk 2]
```diff
@@ -129,13 +129,17 @@ export class VersionsDataSource {
       return pkg.version
     }
 
+    const prefs = await this.ctx.localSettingsApi.getPreferences()
+    const notificationPreferences = prefs.notifyWhenRunCompletes ?? []
```
Comment (warrensplayer) [inline packages/data-context/src/sources/VersionsDataSource.ts]: Does not include when notification set for `starts` or `startsFailing`

File: packages/server/lib/open_project.ts [hunk 3]
```diff
@@ -238,14 +238,16 @@ export class OpenProject {
...
-  changeUrlToDebug () {
+  changeUrlToDebug (runNumber: number) {
     if (!this.projectBase) {
       debug('No projectBase, cannot change url')
 
       return
     }
 
-    const newUrl = `#/debug`
+    const params = JSON.stringify({ from: 'notification', runNumber })
```
Comment (lmiller1990) [inline packages/server/lib/open_project.ts:248]: Neat idea to pass it through here.

**Review by lmiller1990 — COMMENTED** [review]
This looks much cleaner now, thank you @warrensplayer.

It looks like we collect a lot less telemetry than initially - great to see. This looks good to me. I gave it a small test. Let's merge this up and move onto the final QA / acceptance testing.

**Pull request merged**